lfs/batcher_test.go

```diff
@@ -7,28 +7,66 @@ import (
 )
 
 func TestBatcherSizeMet(t *testing.T) {
-	b := NewBatcher(2)
+	assertAll([]batcherTestCase{
+		{2, 4, false},
+		{3, 5, false},
+		{0, 0, false},
+	}, t)
+}
+
+func TestBatcherExit(t *testing.T) {
+	assertAll([]batcherTestCase{
+		{2, 4, true},
+		{3, 5, true},
+		{0, 0, true},
+	}, t)
+}
+
+// Type batcherTestCase specifies information about how to run a particular test
+// around the type lfs.Batcher.
+type batcherTestCase struct {
+	BatchSize  int
+	ItemCount  int
+	ShouldExit bool
+}
 
-	for i := 0; i < 4; i++ {
-		b.Add(&Downloadable{})
+// Func Batcher makes and retunrs a lfs.Batcher according to the specification
+// given in this instance of batcherTestCase. When returned, it is filled with
+// the given amount of items, and has exited if it was told to.
+func (b batcherTestCase) Batcher() *Batcher {
+	batcher := NewBatcher(b.BatchSize)
+	for i := 0; i < b.ItemCount; i++ {
+		batcher.Add(&Downloadable{})
 	}
 
-	group := b.Next()
-	assert.Equal(t, 2, len(group))
+	if b.ShouldExit {
+		batcher.Exit()
+	}
 
-	group = b.Next()
-	assert.Equal(t, 2, len(group))
+	return batcher
 }
 
-func TestBatcherExit(t *testing.T) {
-	b := NewBatcher(4)
-
-	for i := 0; i < 2; i++ {
-		b.Add(&Downloadable{})
+// Func Batches returns the number of individual batches expected to be
+// processed by the batcher under test.
+func (b batcherTestCase) Batches() int {
+	if b.BatchSize == 0 {
+		return 0
 	}
 
-	b.Exit()
+	return b.ItemCount / b.BatchSize
+}
+
+// Func assertAll processes all test cases, throwing assertion errors if they
+// fail.
+func assertAll(cases []batcherTestCase, t *testing.T) {
```

Comment:
I'd like to see a more descriptive name here, like `runBatcherTests`. A name like `assertAll` seems too generic here. This seems to be more than just a simple assertion, as well.